macOS dynamic framework unit tests fail (crash) due to missing token, although a token is supplied

When running the tests of the dynamic framework on macOS, I seem to always run into this error, which results in a crash of the test suite:
```
libc++abi.dylib: terminating with uncaught exception of type std::runtime_error: You must provide a Mapbox API access token for Mapbox tile sources
```
I've tried putting my token in the test bundle's Info.plist as well as in the target's environment. Neither seems to be working. Also, it seems to be a side-effect of a test which has already finished executing, not the test name displayed at the end of the log (which is often different)

I got this in the console, which might be a clue:

```
2017-12-20 23:32:02.624021-0800 xctest[21217:11455112] [ERROR] {com.mapbox.mbgl.Map Snapshotter}[Setup]: loading style failed: HTTP status code 401
```

--- a/platform/darwin/src/MGLImageSource.h
+++ b/platform/darwin/src/MGLImageSource.h
@@ -38,11 +38,8 @@ MGL_EXPORT
    bottomLeft: CLLocationCoordinate2D(latitude: 37.936, longitude: -80.425),
    bottomRight: CLLocationCoordinate2D(latitude: 37.936, longitude: -71.516),
    topRight: CLLocationCoordinate2D(latitude: 46.437, longitude: -71.516))
- let source = MGLImageSource(identifier: "radar-source", coordinateQuad: coordinates, url: URL(string: "https://www.mapbox.com/mapbox-gl-js/assets/radar.gif")!)
+ let source = MGLImageSource(identifier: "radar", coordinateQuad: coordinates, url: URL(string: "https://www.mapbox.com/mapbox-gl-js/assets/radar.gif")!)
  mapView.style?.addSource(source)
- 
- let layer = MGLRasterStyleLayer(identifier: "radar-layer", source: source)
- style.addLayer(layer)
  ```
  */
 MGL_EXPORT

--- a/platform/darwin/src/MGLMapSnapshotter.h
+++ b/platform/darwin/src/MGLMapSnapshotter.h
@@ -145,11 +145,11 @@ typedef void (^MGLMapSnapshotCompletionHandler)(MGLMapSnapshot* _Nullable snapsh
  
  let snapshotter = MGLMapSnapshotter(options: options)
  snapshotter.start { (snapshot, error) in
-     if error != nil {
-         // error handler
-     } else {
-         // image handler
+     if let error = error {
+         fatalError(error.localizedDescription)
      }
+     
+     image = snapshot?.image
  }
  ```
  */

--- a/platform/darwin/test/MGLDocumentationExampleTests.swift
+++ b/platform/darwin/test/MGLDocumentationExampleTests.swift
@@ -27,11 +27,11 @@ import Mapbox
 class MGLDocumentationExampleTests: XCTestCase, MGLMapViewDelegate {
     var mapView: MGLMapView!
     var styleLoadingExpectation: XCTestExpectation!
+    static let styleURL = Bundle(for: MGLDocumentationExampleTests.self).url(forResource: "one-liner", withExtension: "json")!
 
     override func setUp() {
         super.setUp()
-        let styleURL = Bundle(for: MGLDocumentationExampleTests.self).url(forResource: "one-liner", withExtension: "json")
-        mapView = MGLMapView(frame: CGRect(x: 0, y: 0, width: 256, height: 256), styleURL: styleURL)
+        mapView = MGLMapView(frame: CGRect(x: 0, y: 0, width: 256, height: 256), styleURL: MGLDocumentationExampleTests.styleURL)
         mapView.delegate = self
         styleLoadingExpectation = expectation(description: "Map view should finish loading style")
         waitForExpectations(timeout: 1, handler: nil)
@@ -279,6 +279,27 @@ class MGLDocumentationExampleTests: XCTestCase, MGLMapViewDelegate {
     }
     
     func testMGLMapSnapshotter() {
+        let expectation = self.expectation(description: "MGLMapSnapshotter should produce a snapshot")
+        #if os(macOS)
+            var image: NSImage? {
+                didSet {
+                    expectation.fulfill()
+                }
+            }
+        #else
+            var image: UIImage? {
+                didSet {
+                    expectation.fulfill()
+                }
+            }
+        #endif
+        
+        class MGLStyle {
+            static func satelliteStreetsStyleURL() -> URL {
+                return MGLDocumentationExampleTests.styleURL
+            }
+        }
+        
         //#-example-code
         let camera = MGLMapCamera(lookingAtCenter: CLLocationCoordinate2D(latitude: 37.7184, longitude: -122.4365), fromDistance: 100, pitch: 20, heading: 0)
 
@@ -287,13 +308,15 @@ class MGLDocumentationExampleTests: XCTestCase, MGLMapViewDelegate {
 
         let snapshotter = MGLMapSnapshotter(options: options)
         snapshotter.start { (snapshot, error) in
-            if error != nil {
-                // error handler
-            } else {
-                // image handler
+            if let error = error {
+                fatalError(error.localizedDescription)
             }
+            
+            image = snapshot?.image
         }
         //#-end-example-code
+        
+        wait(for: [expectation], timeout: 1)
     }
     
     // For testMGLMapView().

